### ISSUE TITLE
The BillingDataSource is not compatible of the new V6 SUBS

### ISSUE THREAD

**Issue by huajianjiang** [android/play-billing-samples#623]
One subscription product can have many base plans, and each base plan can have many offers:
![image](https://github.com/android/play-billing-samples/assets/12984573/8ea1ed1a-bff8-455d-8c29-caa60f55ebde)
But the `Purchase` returned from `queryPurchasesAsync` method only include `List<String> products` as known as the list of products, old version is list of skus, but not have `basePlanId` and `offerId` field, it cause I can't determine which  base plan or offer was subscribed, and can't update the state of product UI.

**Comment by matale** [comment]
Yeah its limited, see https://stackoverflow.com/questions/72351253/google-play-billing-library-determine-base-plan-expiry-of-subscription-purcha/72673387#72673387

**Comment by huajianjiang** [comment]
@matale It have been more than 1 year, I confused that why Google billing library haven't realized and fixed this problem.
